sdk/storage/azure-storage-blob/azure/storage/blob/_shared/models.py

```diff
@@ -367,15 +330,34 @@ def __str__(self):
                 ('u' if self.update else '') +
                 ('p' if self.process else ''))
 
-
-AccountPermissions.READ = AccountPermissions(read=True)
-AccountPermissions.WRITE = AccountPermissions(write=True)
-AccountPermissions.DELETE = AccountPermissions(delete=True)
-AccountPermissions.LIST = AccountPermissions(list=True)
-AccountPermissions.ADD = AccountPermissions(add=True)
-AccountPermissions.CREATE = AccountPermissions(create=True)
-AccountPermissions.UPDATE = AccountPermissions(update=True)
-AccountPermissions.PROCESS = AccountPermissions(process=True)
+    @classmethod
+    def from_string(cls, permission):
+        if len(permission) > 8:
+            raise ValueError("Invalid Permission String")
+        read, write, delete, list = False, False, False, False
+        add, create, update, process = False, False, False, False
+        curr = 0
+        while curr < len(permission):
+            if permission[curr] =='r':
+                read = True
+            elif permission[curr] == 'w':
+                write = True
+            elif permission[curr] == 'd':
+                delete = True
+            elif permission[curr] == 'l':
+                list = True
+            elif permission[curr] =='a':
+                add = True
+            elif permission[curr] == 'c':
+                create = True
+            elif permission[curr] == 'u':
+                update = True
+            elif permission[curr] == 'p':
+                process = True
+            else:
+                raise ValueError("Invalid Permission String")
```

Comment:
We don't want to raise here, as we want to be able to support strings from future API versions that may have additional letters. Instead, we parse what we can, but keep the original string around and use that for the actual query.
```
def from_string(cls, permissions):
    # parse string
    parsed = cls(read=True, write=False, ....)
    parsed._str = permissions  # keeping original value
    return parsed
````
```